tools/executors/translate/write.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  appendFileSync,
+  existsSync,
+  mkdirSync,
+  unlinkSync,
+  writeFileSync,
+} from 'fs';
+import { basename, join } from 'path';
+import * as ts from 'typescript';
+import { UNICODE_REGEXP } from './constants';
+import { EchoExecutorOptions } from './options';
+import { translateText } from './translate';
+
+export function writeExportStatements(
+  options: EchoExecutorOptions,
+  translationIndex: string
+) {
+  for (const language of options.languages) {
+    console.info(
+      `Appending export statement for language ${language} in ${translationIndex}`
+    );
+    const outputDir = join(options.output, language);
+    const languageIndex = join(outputDir, `index.ts`);
+    if (existsSync(outputDir) && existsSync(languageIndex)) {
+      const exportStatement = `// THIS FILE IS AUTO-GENERATED THROUGH \`npm run translate\`
+export * as ${language} from './${language}';\n`;
+      appendFileSync(translationIndex, exportStatement, {
+        encoding: 'utf-8',
+      });
+    }
+  }
+}
+
+export async function writeTranslation(
+  options: EchoExecutorOptions,
+  language: string,
+  stringLiterals: string[],
+  sourceFile: ts.SourceFile,
+  printer: ts.Printer,
+  filePath: string
+) {
+  const outputDir = join(options.output, language);
+
+  const tText = await translateText(stringLiterals, language);
+
+  const result = ts.transform(sourceFile, [translateTransformer(tText)]);
+
+  let content = printer.printFile(result.transformed[0]);
+
+  // convert unicodes to string
+  content = content.replace(UNICODE_REGEXP, (match, grp) => {
+    return String.fromCharCode(parseInt(grp, 16));
+  });
+
+  content =
+    `// THIS FILE IS AUTO-GENERATED THROUGH \`npm run translate\`\n` + content;
+
+  const fileNameWExt = basename(filePath);
+
+  const outputFilePath = join(outputDir, fileNameWExt);
+
+  console.info(`Writing translated file at: ${outputFilePath}`);
+
+  // Write pretty printed transformed typescript to output directory
+  writeFileSync(outputFilePath, content);
+
+  const index = join(outputDir, `index.ts`);
+
+  const fileNameWOExt = basename(filePath, '.ts');
+
+  const exportStatement = `// THIS FILE IS AUTO-GENERATED THROUGH \`npm run translate\`
+export * from './${fileNameWOExt}';\n`;
+
+  console.info(`Writing export statement at: ${index}`);
+
+  appendFileSync(index, exportStatement, { encoding: 'utf-8' });
+}
+
+export function deleteLanguageIndexFiles(options: EchoExecutorOptions) {
+  for (const language of options.languages) {
+    console.info(`Deleting index file for language: ${language}`);
+    const outputDir = join(options.output, language);
+    if (!existsSync(outputDir)) {
+      mkdirSync(outputDir);
```

Comment:
The name of the function is `delete`, but you're creating it here.

Reply:
Renamed function name